streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -62,9 +60,9 @@ public static StoreFactory create(final String name) {
                                     @Override
                                     public StateStoreSupplier build() {
                                         if (capacity < Integer.MAX_VALUE) {
-                                            return new InMemoryLRUCacheStoreSupplier<>(name, capacity, serdes, null);
+                                            return new InMemoryLRUCacheStoreSupplier<>(name, capacity, keySerde, valueSerde, null);
                                         }
-                                        return new InMemoryKeyValueStoreSupplier<>(name, serdes, null);
+                                        return new InMemoryKeyValueStoreSupplier<>(name, keySerde, valueSerde, null);
```

Comment:
Another option, perhaps better option, would be to have an additional constructor to `InMemoryLRUCacheStoreSupplier` and to similar `null`-affected classes such as `RocksDBWindowStoreSupplier` that define the "ignore time" contract.  Idea:

```java
    /**
     * Returns a store supplier that ignores time.
     */
    public InMemoryLRUCacheStoreSupplier(String name, int capacity, StateSerdes<K, V> serdes, Time time) {
        this(name, capacity, serdes, null);
    }
```